src/metrics/rules/active_ruleset.go

```diff
@@ -534,6 +534,7 @@ func (as *activeRuleSet) matchRollupTarget(
 		nameTagValue    []byte
 		includeTagNames = as.includeTagKeys
 	)
+	defer sortedTagIter.Close()
```

Comment:
Nit: let's change this slightly so that the allocation is right next to the `Close()`:

```
sortedTagIter  := matchOpts.SortedTagIteratorFn(sortedTagPairBytes)
defer sortedTagIter.Close()
```

In practice this won't matter all, just very slightly more standard I think.